pkg/addons/manifest.go

```diff
@@ -295,10 +296,33 @@ func txtFuncMap(overwriteRegistry string) template.FuncMap {
 	funcs["caBundleVolumeMount"] = caBundleVolumeMountTemplateFunc
 	funcs["EquinixMetalSecret"] = equinixMetalSecretTemplateFunc
 	funcs["vSphereCSIWebhookConfig"] = vSphereCSIWebhookConfigTemplateFunc
+	funcs["awsCCMCloudConfig"] = awsCCMCloudConfigTemplateFunc
 
 	return funcs
 }
 
+func awsCCMCloudConfigTemplateFunc(name string, ipFamily kubeoneapi.IPFamily) string {
+	lines := []string{
+		"[global]",
+		fmt.Sprintf("KubernetesClusterID=%q", name),
+	}
+
+	switch ipFamily {
+	case kubeoneapi.IPFamilyIPv4:
+		lines = append(lines, fmt.Sprintf("NodeIPFamilies=%q", "ipv4"))
+	case kubeoneapi.IPFamilyIPv6:
+		lines = append(lines, fmt.Sprintf("NodeIPFamilies=%q", "ipv6"))
+	case kubeoneapi.IPFamilyIPv4IPv6:
+		lines = append(lines, fmt.Sprintf("NodeIPFamilies=%q", "ipv4"))
+		lines = append(lines, fmt.Sprintf("NodeIPFamilies=%q", "ipv6"))
```

Comment:
Note for future readers: I checked to make sure it's possible to specify `NodeIPFamilies` multiple times like this and it is. Here's an example I found in the AWS CCM tests: https://github.com/kubernetes/cloud-provider-aws/blob/15fa84711702e2a57997c934d2963b29284f8bdc/pkg/providers/v1/aws_test.go#L238